### DISCUSSION TITLE
How to change the repositories to local

### DISCUSSION THREAD

**Discussion by mxwmwx** [dbeaver/dbeaver#7275] [DBeaver Development] [answered]
After a package is completed, the next package may have some configuration changes, such as the previous jsqlparser and jexl2, these changes cause the code on the old version to rely on jexl2 and jsqlparser1.4, but the latest is jexl3 and jsqlparser_3. 0.0. How to set the repositories to be local, not your official cloud. . I want to change the relevant dependency download path to local, so I can always try to develop some other features based on a version without worrying about packaging. Is there any related documentation explaining how to modify related files? Or would you like to tell me how to change the configuration? Just give an example.Thanks!

**Comment by serge-rider** [comment, marked as answer]
In fact dbeaver depends on jsqlparser 3.2 and jexl 3.1. Make sure you use branch `devel` (master is deprecated).

Generally you can change property `local-p2-repo.url` in root pom.xml. It refers to P2 repository with 3rd party dependencies. The list of dependencies is located in https://github.com/dbeaver/dbeaver/blob/devel/product/localRepository/pom.xml

You can create your own P2 repository and overwrite local-p2-repo.url (e.g. with Maven cli parameters).
